code/modules/reagents/reagent_containers/patch.dm

```diff
@@ -13,6 +13,16 @@
 	..()
 	icon_state = "bandaid" // thanks inheritance
 
+/obj/item/weapon/reagent_containers/pill/patch/attack(mob/living/M, mob/user)
+	if(!canconsume(M, user))
```

Comment:
goof is right surprisingly
the reason this is runtiming is because you typecast with no sanity and has absolutely nothing to do with this proc which should not in fact not be here